### ISSUE TITLE
Invalid API key/secret pair returned when calling createLoanOffer

### ISSUE THREAD

**Issue by DaveDubUK** [premasagar/poloniex.js#6]
The key/secret pair is known to be good, as correct values are returned when poloniex.returnCompleteBalances is called.

However, when a call to createLoanOffer is made (using the same key/secret pair), "error": "Invalid API key/secret pair." is returned.

This could be an issue with Poloniex (I have already notified them of the problem)

**Comment by DaveDubUK** [comment]
Update: I had a reply for Poloniex:
<quote>
I understand, are you certain the same code that create the headers for the other calls is being used for the loan order call? There is no other instance where this error is called other than actual wrong API keys and the server expect the same kind of header as other commands, there is no difference between createLoanOffer and any other private call.

I personally am using this API command and encountering no issue with it, I am afraid there may be something out of place on the code possibly.
</quote>

I've looked through the relevant poloniex.js code myself and cannot see any difference in the way the createLoanOffer call is made (i.e. via the _private call) and am still at a loss.

Any help gratefully received!

Kind Regards,

Davedub

**Comment by DaveDubUK** [comment]
Ok, this is officially driving me nuts!

I have minimised my code to something very simple and experimented. I have found that all calls that do not require extra parameters will always work fine. All calls made WITH parameters fail consistently. Here's my code (I have not shown my key/secret pair for obvious reasons):

############################################################
// Poloniex API boilerplate
var Poloniex = require('./lib/poloniex'); // p.s. using poloniex.js here as suggested doesn't work!
var poloniexKey = "<<< hidden! >>>"
var poloniexSecret = "<<< hidden! >>>";
// Create a new instance, with optional API key and secret
poloniex = new Poloniex(
	poloniexKey,
	poloniexSecret
);
// end Poloniex API boilerplate

poloniex.cancelLoanOffer(87794586, function(err, data) {
    if (err) {
        console.log("PoloniexBot: Error: " + err);
        return;
    } else {
    	var issue = "no problems";
    	for (item in data) {
    		    //console.log("Item: " + item + "\n" + JSON.stringify(data[item], null, " "));
    		    if (item == "error") issue = JSON.stringify(data[item], null, " ");
		}
		console.log(issue);		
	}
});
############################################################

The code above will not work, and the 'issue' variable will contain "Invalid API key/secret pair."
The code below will work fine, as will any of the calls that do not require parameters:

############################################################
// Poloniex API boilerplate
var Poloniex = require('./lib/poloniex'); // p.s. using poloniex.js here as suggested doesn't work!
var poloniexKey = "<<< hidden! >>>"
var poloniexSecret = "<<< hidden! >>>";
// Create a new instance, with optional API key and secret
poloniex = new Poloniex(
	poloniexKey,
	poloniexSecret
);
// end Poloniex API boilerplate

poloniex.returnCompleteBalances(function(err, data) {
    if (err) {
        console.log("PoloniexBot: Error: " + err);
        return;
    } else {
    	var issue = "no problems";
    	for (item in data) {
    		    //console.log("Item: " + item + "\n" + JSON.stringify(data[item], null, " "));
    		    if (item == "error") issue = JSON.stringify(data[item], null, " ");
		}
		console.log(issue);		
	}
});
############################################################

Something is not right with calls made using parameters - any ideas?

 - Davedub

**Comment by premasagar** [comment]
Thanks @DaveDubUK. I'm not sure the source of this issue. @anders94 do you have any insight?

**Comment by anders94** [comment]
@DaveDubUK did you get the module via npm install or is this the git checkout?

**Comment by DaveDubUK** [comment]
Hi Anders,

Thanks for getting in touch. I've actually tried both ways, same result
either way.

- Davedub

**Comment by anders94** [comment]
@DaveDubUK I think I found the issue. Somewhat subtle bug with a sort operation. Please give this PR a try: https://github.com/premasagar/poloniex.js/pull/8

**Comment by dutu** [comment]
@DaveDubUK, did you try passing the orderId parameter as string instead of integer?

**Comment by anders94** [comment]
I'm going to merge this PR and close this issue because this update fixes this bug.

**Comment by DaveDubUK** [comment]
Apologies for not getting back on this earlier, I have been away on a short break for a few days.

I have completed initial tests on this and can confirm the issue is fixed.

Many thanks @anders94, very much appreciated!

**Comment by ElGoorf** [comment]
Hi guys, I've just ran into this exact same problem. I can run other functions perfectly fine, I'm seeing all my private data (active loans, etc), but when I call `createLoanOffer` I get `error: 'Invalid API key/secret pair.'`

nove v10.20.1

package.json:
```
{
...
  "dependencies": {
    "asciichart": "^1.5.25",
    "poloniex.js": "^1.0.1"
  }
}
```

Code: `poloniex.createLoanOffer("USDT", 1000, 2, 0, 0.0006, console.log);`

**Comment by anders94** [comment]
It is possible they "fixed" the parameter ordering which breaks the fix in #8 - could try reversing that change.
